frontend/src/store/slices/userSlice.ts

```diff
@@ -32,19 +33,45 @@ export const getUserDetails =
 		dispatch(slice.actions.addUser(apiUserToStore(user)));
 	};
 
+export const createNewArtistThunk = createAsyncThunk(
+	"users/createArtist",
+	async (artist: FormData, { dispatch, getState }) => {
+		try {
+			const artistUser = await api.artists.update(artist);
+			console.log("artist user: ", artistUser);
+			const currentState = getState() as RootState;
+			const currentUser = currentState.session.user!;
+			const { id, username } = currentUser!;
+			const newArtist = {
+				...artistUser,
+				id,
+				stage_name: artistUser.stage_name || username,
+			};
+			dispatch(slice.actions.updateUser(apiUserToStore(newArtist)));
+		} catch (e) {
+			if (e instanceof Error) {
+				return e.api;
+			}
+			throw e;
+		}
+	},
+);
+
 export const slice = createSlice({
 	name: "users",
 	initialState,
 	reducers: {
 		addUser: (store, action: PayloadAction<User>) => {
 			store.users[action.payload.id] = action.payload;
 		},
-
 		addUsers: (store, action: PayloadAction<User[]>) => {
 			for (const u of action.payload) {
 				store.users[u.id] = u;
 			}
 		},
+		updateUser: (store, action: PayloadAction<User>) => {
+			store.users[action.payload.id] = action.payload;
+		},
```

Comment:
This definition is exactly equivalent to addUser, shouldn't we remove it and have things calling updateUser call addUser instead? we could also pick a better name for addUser to encapsulate both uses, but having two identical functions can increase maintainability burdens

Reply:
that's what being tired does!  I see they're exactly the same now.  I see addUser is setting a user, which is exactly what we need to do when updating a user.